bigtable/admin.go

```diff
@@ -2150,9 +2150,60 @@ func (ac *AdminClient) RestoreTableFrom(ctx context.Context, sourceInstance, tab
 	return longrunning.InternalNewOperation(ac.lroClient, op).Wait(ctx, &resp)
 }
 
+type backupOptions struct {
+	backupType        *BackupType
+	hotToStandardTime *time.Time
+}
+
+// BackupOption can be used to specify parameters for backup operations.
+type BackupOption interface {
+	apply(o *backupOptions)
+}
+
+// setHotToStandardTime can be used to set backup type to [BackupTypeHot] and the
+// time at which the hot backup will
+// be converted to a standard backup.
+// Once the `hot_to_standard_time` has passed, Cloud Bigtable will convert the
+// hot backup to a standard backup. This value must be greater than the backup
+// creation time by at least 24 hours.
+type hotBackupOption struct {
```

Comment:
do we need these `hotBackupOption` lines and functions?

I'm wondering how useful these will be, as opposed to a user creating their own `backupOptions` to create a hot backup

Reply:
hotBackupOption is a private struct and not visible to the user. Having this simplifies the user experience. To create a hot backup, user can just do below instead of having to create their own struct. 
```
hotToStdTime := time.Now()
adminClient.CreateBackupWithOptions(ctx, "TableID", "Cluster", "BkpName",
		time.Now().Add(24 * 4 * time.Hour), HotBackup())
```

This is the common pattern followed across libraries in google-cloud-go to create optional parameters e.g.

https://github.com/googleapis/google-cloud-go/blob/649c075d5310e2fac64a0b65ec445e7caef42cb0/storage/storage.go#L2147-L2221

Reply:
alright, I'll defer to the readability reviewer